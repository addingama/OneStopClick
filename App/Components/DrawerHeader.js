import React, { Component, PropTypes } from 'react'
import { Header } from 'react-native-elements'
import { HamburgerMenu } from './FormGenerator'
import styles from './Styles/DrawerHeaderStyle'

export default class DrawerHeader extends Component {
  openMenu () {
    this.props.navigation.navigate('DrawerOpen')
  }

  render () {
    const { title, rightComponent } = this.props
    return (
      <Header
        backgroundColor='#2F1F37'
        leftComponent={<HamburgerMenu onPress={this.openMenu.bind(this)} />}
        centerComponent={{ text: title, style: styles.titleStyle }}
        rightComponent={rightComponent}
      />
    )
  }
}

DrawerHeader.propTypes = {
  title: PropTypes.string.isRequired,
  navigation: PropTypes.object.isRequired
}
