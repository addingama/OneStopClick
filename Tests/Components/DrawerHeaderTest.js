import 'react-native'
import React from 'react'
import DrawerHeader from '../../App/Components/DrawerHeader'
import renderer from 'react-test-renderer'

test('DrawerHeader component renders correctly', () => {
  const tree = renderer.create(<DrawerHeader title='Drawer title' />).toJSON()
  expect(tree).toMatchSnapshot()
})
