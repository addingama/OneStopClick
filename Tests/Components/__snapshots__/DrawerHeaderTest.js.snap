// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerHeader component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 1,
        "height": 70,
        "left": 0,
        "padding": 15,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      Object {
        "backgroundColor": "#2F1F37",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        undefined,
      ]
    }
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          onPress={undefined}
          style={
            Array [
              false,
              false,
              false,
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          underlayColor="white"
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 28,
                },
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ],
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            î—’
          </Text>
        </View>
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Drawer title
    </Text>
    <View
      style={
        Object {
          "height": 24,
          "width": 24,
        }
      }
    />
  </View>
</View>
`;
